### PR TITLE
HBASE-22510 Address findbugs/spotbugs complaints (branch-1)

### PR DESCRIPTION
**Pull request by apurtell**

### CODE REVIEW

File: hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/EncodedDataBlock.java [hunk 1]
```diff
@@ -255,7 +255,7 @@ private ByteBuffer getUncompressedBuffer() {
       }
       BufferGrabbingByteArrayOutputStream stream = new BufferGrabbingByteArrayOutputStream();
       baos.writeTo(stream);
-      this.dataBlockEncoder.endBlockEncoding(encodingCtx, out, stream.getOurBytes());
+      this.dataBlockEncoder.endBlockEncoding(encodingCtx, out, stream.toByteArray());
```
Comment (apurtell) [inline hbase-common/src/main/java/org/apache/hadoop/hbase/io/encoding/EncodedDataBlock.java]: Changes in this file keep the code in sync with changes proposed on #277

File: hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java [hunk 2]
```diff
@@ -7959,7 +7959,9 @@ public Result append(Append mutate, long nonceGroup, long nonce) throws IOExcept
         for (Map.Entry<Store, List<Cell>> entry: removedCellsForMemStore.entrySet()) {
           entry.getKey().add(entry.getValue());
         }
-        if (we != null) mvcc.complete(we);
```
Comment (apurtell) [inline hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java]: Changes to this file are a carrying forward of formatting changes from the branch-1.3 changes proposed on #275 and #277. They don't address a findbugs issue, only try to keep the code as consistent as possible between branches.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 28 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | findbugs | 1 | Findbugs executables are not available. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ branch-1 Compile Tests _ |
| 0 | mvndep | 59 | Maven dependency ordering for branch |
| +1 | mvninstall | 120 | branch-1 passed |
| +1 | compile | 53 | branch-1 passed with JDK v1.8.0_212 |
| +1 | compile | 61 | branch-1 passed with JDK v1.7.0_222 |
| +1 | checkstyle | 112 | branch-1 passed |
| +1 | shadedjars | 161 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 48 | branch-1 passed with JDK v1.8.0_212 |
| +1 | javadoc | 61 | branch-1 passed with JDK v1.7.0_222 |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 14 | Maven dependency ordering for patch |
| +1 | mvninstall | 103 | the patch passed |
| +1 | compile | 54 | the patch passed with JDK v1.8.0_212 |
| +1 | javac | 54 | the patch passed |
| +1 | compile | 61 | the patch passed with JDK v1.7.0_222 |
| +1 | javac | 61 | the patch passed |
| +1 | checkstyle | 26 | The patch passed checkstyle in hbase-common |
| +1 | checkstyle | 85 | hbase-server: The patch generated 0 new + 595 unchanged - 6 fixed = 595 total (was 601) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 159 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 241 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2. |
| +1 | javadoc | 47 | the patch passed with JDK v1.8.0_212 |
| +1 | javadoc | 58 | the patch passed with JDK v1.7.0_222 |
||| _ Other Tests _ |
| +1 | unit | 151 | hbase-common in the patch passed. |
| +1 | unit | 6140 | hbase-server in the patch passed. |
| +1 | asflicense | 40 | The patch does not generate ASF License warnings. |
| | | 8075 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-278/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/278 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux bfd142b328b2 4.4.0-138-generic #164-Ubuntu SMP Tue Oct 2 17:16:02 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | branch-1 / e233cfb |
| maven | version: Apache Maven 3.0.5 |
| Default Java | 1.7.0_222 |
| Multi-JDK versions |  /usr/lib/jvm/java-8-openjdk-amd64:1.8.0_212 /usr/lib/jvm/java-7-openjdk-amd64:1.7.0_222 |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-278/1/testReport/ |
| Max. process+thread count | 4669 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-278/1/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by apurtell** [comment]
Precommit on JIRA was good: https://issues.apache.org/jira/browse/HBASE-22510?focusedCommentId=16852606&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-16852606

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 49 | Docker mode activated. |
||| _ Prechecks _ |
| 0 | findbugs | 1 | Findbugs executables are not available. |
| +1 | hbaseanti | 0 |  Patch does not have any anti-patterns. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -0 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ branch-1 Compile Tests _ |
| 0 | mvndep | 88 | Maven dependency ordering for branch |
| +1 | mvninstall | 107 | branch-1 passed |
| +1 | compile | 55 | branch-1 passed with JDK v1.8.0_212 |
| +1 | compile | 61 | branch-1 passed with JDK v1.7.0_222 |
| +1 | checkstyle | 117 | branch-1 passed |
| +1 | shadedjars | 173 | branch has no errors when building our shaded downstream artifacts. |
| +1 | javadoc | 46 | branch-1 passed with JDK v1.8.0_212 |
| +1 | javadoc | 60 | branch-1 passed with JDK v1.7.0_222 |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 13 | Maven dependency ordering for patch |
| +1 | mvninstall | 104 | the patch passed |
| +1 | compile | 53 | the patch passed with JDK v1.8.0_212 |
| +1 | javac | 54 | the patch passed |
| +1 | compile | 61 | the patch passed with JDK v1.7.0_222 |
| +1 | javac | 61 | the patch passed |
| +1 | checkstyle | 25 | The patch passed checkstyle in hbase-common |
| +1 | checkstyle | 92 | hbase-server: The patch generated 0 new + 595 unchanged - 6 fixed = 595 total (was 601) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedjars | 174 | patch has no errors when building our shaded downstream artifacts. |
| +1 | hadoopcheck | 242 | Patch does not cause any errors with Hadoop 2.8.5 2.9.2. |
| +1 | javadoc | 47 | the patch passed with JDK v1.8.0_212 |
| +1 | javadoc | 60 | the patch passed with JDK v1.7.0_222 |
||| _ Other Tests _ |
| +1 | unit | 135 | hbase-common in the patch passed. |
| -1 | unit | 8092 | hbase-server in the patch failed. |
| +1 | asflicense | 48 | The patch does not generate ASF License warnings. |
| | | 10098 | |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.coprocessor.TestMetaTableMetrics |
|   | hadoop.hbase.client.TestSnapshotWithAcl |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=17.05.0-ce Server=17.05.0-ce base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-278/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/278 |
| Optional Tests |  dupname  asflicense  javac  javadoc  unit  findbugs  shadedjars  hadoopcheck  hbaseanti  checkstyle  compile  |
| uname | Linux 1dd9425c8fdb 4.4.0-141-generic #167~14.04.1-Ubuntu SMP Mon Dec 10 13:20:24 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /testptch/patchprocess/precommit/personality/provided.sh |
| git revision | branch-1 / db6ad39 |
| maven | version: Apache Maven 3.0.5 |
| Default Java | 1.7.0_222 |
| Multi-JDK versions |  /usr/lib/jvm/java-8-openjdk-amd64:1.8.0_212 /usr/lib/jvm/java-7-openjdk-amd64:1.7.0_222 |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-278/2/artifact/out/patch-unit-hbase-server.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-278/2/testReport/ |
| Max. process+thread count | 3847 (vs. ulimit of 10000) |
| modules | C: hbase-common hbase-server U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-278/2/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 0 | Docker mode activated. |
| -1 | patch | 8 | https://github.com/apache/hbase/pull/278 does not apply to branch-1. Rebase required? Wrong Branch? See https://yetus.apache.org/documentation/in-progress/precommit-patchnames for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| GITHUB PR | https://github.com/apache/hbase/pull/278 |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-278/3/console |
| Powered by | Apache Yetus 0.9.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request merged**